kratos/modified_shape_functions/modified_shape_functions.h

```diff
@@ -0,0 +1,330 @@
+    * @return rNegativeSideShapeFunctionValues: Matrix containing the negative side computed shape function values.
+    * @return rNegativeSideShapeFunctionsGradientsValues: std::vector containing the shape functions gradients values on the negative side.
+    * @return rNegativeSideWeightsValues: Vector containing the Gauss pts. negative side weights (already multiplied by the Jacobian).
+    * @param IntegrationMethod: Desired integration quadrature.
+    */
+    virtual void GetNegativeSideShapeFunctionsAndGradientsValues(
+        Matrix &rNegativeSideShapeFunctionsValues,
+        std::vector<Matrix> &rNegativeSideShapeFunctionsGradientsValues,
+        Vector &rNegativeSideWeightsValues,
+        const IntegrationMethodType IntegrationMethod);
+
+    /**
+    * Returns the shape function values in the positive split element interface side for a given quadrature.
+    * @return rInterfacePositiveSideShapeFunctionValues: Matrix containing the positive side computed shape function values.
+    * @return rInterfacePositiveSideShapeFunctionsGradientsValues: std::vector containing the shape functions gradients values on the positive side.
+    * @return rInterfacePositiveSideWeightsValues: Vector containing the Gauss pts. positive side weights (already multiplied by the Jacobian).
+    * @param IntegrationMethod: Desired integration quadrature.
+    */
+    virtual void GetInterfacePositiveSideShapeFunctionsAndGradientsValues(
+        Matrix &rInterfacePositiveSideShapeFunctionsValues,
+        std::vector<Matrix> &rInterfacePositiveSideShapeFunctionsGradientsValues,
+        Vector &rInterfacePositiveSideWeightsValues,
+        const IntegrationMethodType IntegrationMethod);
+
+    /**
+    * Returns the shape function values in the negative split element side for a given quadrature.
+    * @return rInterfaceNegativeSideShapeFunctionValues: Matrix containing the negative side computed shape function values.
+    * @return rInterfaceNegativeSideShapeFunctionsGradientsValues: std::vector containing the shape functions gradients values on the negative side.
+    * @return rInterfaceNegativeSideWeightsValues: Vector containing the Gauss pts. negative side weights (already multiplied by the Jacobian).
+    * @param IntegrationMethod: Desired integration quadrature.
+    */
+    virtual void GetInterfaceNegativeSideShapeFunctionsAndGradientsValues(
+        Matrix &rInterfaceNegativeSideShapeFunctionsValues,
+        std::vector<Matrix> &rInterfaceNegativeSideShapeFunctionsGradientsValues,
+        Vector &rInterfaceNegativeSideWeightsValues,
+        const IntegrationMethodType IntegrationMethod);
+        
+    /**
+    * Returns the positive side outwards unit normal vector values for the Gauss pts. of given quadrature.
+    * @return rPositiveSideInterfaceUnitNormal: Outwards unit normal vector list.
+    * @param IntegrationMethod: Desired integration quadrature.
+    */
+    virtual void GetPositiveSideInterfaceUnitNormals(
```

Comment:
same (make it const if you can)

Reply:
note that you will (most probably) be able to make it const if you don't change any of the internals. howevr in this case the method should be a "CalculatePosit...." and not "GEt..."